### PR TITLE
Add proxy logs to logs for VMs

### PR DESCRIPTION
**Pull request by ZhengzheYang**
ref: #25037
This PR extend the dumping logic so that failed VM tests would contain envoy logs along with the logs from the iptables.
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by ZhengzheYang** [comment]
@howardjohn Does this roughly look like what you want? I queried the proxy log the same way as DumpPodProxies. Let me know if I understand the issue wrong and I will fix it!

File: pkg/test/kube/dump.go [hunk 1]
```diff
@@ -129,17 +130,35 @@ func DumpPodEvents(a Accessor, workDir, namespace string, pods ...kubeApiCore.Po
...
 			l, err := a.Logs(pod.Namespace, pod.Name, container.Name, false /* previousLog */)
 			if err != nil {
 				scopes.Framework.Errorf("Unable to get logs for pod/container: %s/%s/%s", pod.Namespace, pod.Name, container.Name)
 				continue
 			}
+			sb.WriteString(l)
+
+			// Get proxy logs if the pod is a VM, since kubectl logs only shows the logs from iptables
+			if isVM && container.Name == "istio-proxy" {
+				if cfgDump, err := a.Exec(pod.Namespace, pod.Name, container.Name, "pilot-agent request GET config_dump"); err == nil {
```
Comment (howardjohn) [inline pkg/test/kube/dump.go:147]: we already capture this somewhere else, what we are looking for is the log written to file. So we can probably do:

```
cat /var/log/istio/istio.err.log
```
and 
```
cat /var/log/istio/istio.log
```
as part of the log

File: pkg/test/kube/dump.go [hunk 2]
```diff
@@ -129,17 +130,35 @@ func DumpPodEvents(a Accessor, workDir, namespace string, pods ...kubeApiCore.Po
...
+				if clustersDump, err := a.Exec(pod.Namespace, pod.Name, container.Name, "pilot-agent request GET clusters"); err == nil {
+					sb.WriteString(clustersDump)
+				} else {
+					scopes.Framework.Errorf("Unable to get istio-proxy clusters for pod: %s/%s", pod.Namespace, pod.Name)
+				}
+			}
 
 			fname := path.Join(workDir, fmt.Sprintf("%s-%s.log", pod.Name, container.Name))
-			if err = ioutil.WriteFile(fname, []byte(l), os.ModePerm); err != nil {
+			if err = ioutil.WriteFile(fname, []byte(sb.String()), os.ModePerm); err != nil {
```
Comment (howardjohn) [inline pkg/test/kube/dump.go:161]: Might make sense to split this into 3 files, to make it bit more clear

File: pkg/test/kube/dump.go (hunk 1 above)
Comment (ZhengzheYang) [reply pkg/test/kube/dump.go:147]: Oh so we actually want the logs from the pod? Understood. Will update!

File: pkg/test/kube/dump.go (hunk 2 above)
Comment (ZhengzheYang) [reply pkg/test/kube/dump.go:161]: Will do! Thanks for pointing it out.

**Pull request merged**